### PR TITLE
DROOLS-5663 additional test suite only for default=exec model

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**:  https://issues.redhat.com/browse/DROOLS-5663

**referenced Pull Requests**: 

pending for https://github.com/kiegroup/droolsjbpm-integration/pull/2247 to merge

then will rebase,
skip 985a295
and submit only 4fdfa93

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tarilabs** [comment]
```
Test Result (8 failures / +8)
org.kie.server.integrationtests.controller.KieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36769/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.KieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36769/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38246/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38246/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38246/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
```

🤷‍♂️ 

Jenkins retest this

File: kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/kjar-with-instrumentation-default-execmodel-wrapper/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.drools.testcoverage</groupId>
+    <artifactId>kie-maven-plugin-default-execmodel-tests-parent</artifactId>
+    <version>7.44.0-SNAPSHOT</version>
```
Comment (tkobayas) [inline kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/kjar-with-instrumentation-default-execmodel-wrapper/pom.xml:8]: Should this be aligned with 7.45.0-SNAPSHOT now?

File: kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.drools.testcoverage</groupId>
+    <artifactId>kie-integration-test-coverage-parent</artifactId>
+    <version>7.44.0-SNAPSHOT</version>
```
Comment (tkobayas) [inline kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/pom.xml:8]: Same here for 7.45.0-SNAPSHOT

File: kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/test-suite/pom.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.drools.testcoverage</groupId>
+    <artifactId>kie-maven-plugin-default-execmodel-tests-parent</artifactId>
+    <version>7.44.0-SNAPSHOT</version>
```
Comment (tkobayas) [inline kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/test-suite/pom.xml:8]: Same here for 7.45.0-SNAPSHOT

File: kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/kjar-with-instrumentation-default-execmodel-wrapper/pom.xml (hunk 1 above)
Comment (tarilabs) [reply kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/kjar-with-instrumentation-default-execmodel-wrapper/pom.xml]: +1

File: kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/pom.xml (hunk 2 above)
Comment (tarilabs) [reply kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/pom.xml]: +1

File: kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/test-suite/pom.xml (hunk 3 above)
Comment (tarilabs) [reply kie-integration-test-coverage/kie-maven-plugin-default-execmodel-tests/test-suite/pom.xml]: +1

**Comment by tarilabs** [comment]
thank you @tkobayas well spotted I updated following yesterday version bump 👍

**Review by tkobayas — APPROVED** [review]

**Review by lucamolteni — APPROVED** [review]

**Comment by tarilabs** [comment]
```
Test Result (8 failures / ±0)
org.kie.server.integrationtests.controller.KieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:42946/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.KieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:42946/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38783/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38783/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38783/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
```

image

🤷‍♂️ 🎱 

Jenkins retest this

**Review by hellowdan — APPROVED** [review]
Thanks a lot, @tarilabs !

**Comment by tarilabs** [comment]
Here's a recap.

Execution https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/droolsjbpm-integration-master.pullrequests/180/ fails with the same 8 failed tests.

```
Test Result (8 failures / ±0)
org.kie.server.integrationtests.controller.KieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38104/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.KieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:38104/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36381/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36381/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36381/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.drools.ActivationGroupIntegrationTest.testClearActivationGroup[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
```

as per also previous run above.

Other PRs currently open see the same failures, for example this one:
https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/droolsjbpm-integration-master.pullrequests/179/
see the same failures.

Also this other run see the same failures:
https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/droolsjbpm-integration-master.pullrequests/175/
image
but is merged anyway, despite the same failures: https://github.com/kiegroup/droolsjbpm-integration/pull/2252#event-3818409639

One can conclude the failures are un-related, already existing on master regardless of this PR, and we should merge this PR likewise the other was merged anyway.

**Review by mariofusco — APPROVED** [review]

**Pull request merged**